app/models/instance.rb

```diff
@@ -9,30 +9,24 @@
 #
 
 class Instance < ApplicationRecord
+  include DatabaseViewRecord
+
   self.primary_key = :domain
 
   attr_accessor :failure_days
 
-  has_many :accounts, foreign_key: :domain, primary_key: :domain, inverse_of: false
-
   with_options foreign_key: :domain, primary_key: :domain, inverse_of: false do
     belongs_to :domain_block
     belongs_to :domain_allow
     belongs_to :unavailable_domain # skipcq: RB-RL1031
+
+    has_many :accounts, dependent: nil
```

Comment:
There was a rubocop violation here on the Rails/HasManyOrHasOneDependent cop. This cop does not run on models that have `readonly?` defined as true, but it looks like when that definition is moved from the model to a concern rubocop does detect it. Arguably this is a false positive to fix at the rubocop level, but I noticed this association has the same options at the various belongs_to associations, so I moved it in there and gave it a `dependent: nil` (default behavior, but more explicit) which should be harmless and satisfy the cop.